### PR TITLE
podman logs: fix tailing

### PR DESCRIPTION
**Pull request by baude**
Fix issues with tailing of container logs as described
in issue #16.  Also add in the ability to use a duration or
known time stamp formats for the --since flag.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podman/logs.go [hunk 1]
```diff
@@ -2,20 +2,24 @@ package main
 
 import (
 	"fmt"
+	"io"
+	"os"
 	"strings"
 	"time"
 
-	"github.com/hpcloud/tail"
+	//"github.com/hpcloud/tail"
```
Comment (mheon) [inline cmd/podman/logs.go]: If this is gone, can you remove it?

File: cmd/podman/logs.go [hunk 2]
```diff
@@ -95,52 +105,116 @@ func logsCmd(c *cli.Context) error {
...
+
+	return nil
+}
+
+func readLog(reader *bufio.Reader, opts logOptions) []string {
+	var output []string
+	for {
+		line, err := reader.ReadString('\n')
+		if err != nil {
+			break
```
Comment (mheon) [inline cmd/podman/logs.go]: Should we check if this is actually EOF or something else blew up?

File: cmd/podman/logs.go [hunk 3]
```diff
@@ -158,3 +232,25 @@ func secondSpaceIndex(line string) int {
 	}
 	return index
 }
+
+// parseInputTime takes the users input and to determine if it is valid and
+// returns a time format and error.  The input is compared to known time formats
+// or a duration which implies now-duration
```
Comment (TomSweeneyRedHat) [inline cmd/podman/logs.go]: now-duration or no-duration?

File: cmd/podman/logs.go [hunk 4]
```diff
@@ -95,52 +105,116 @@ func logsCmd(c *cli.Context) error {
...
+				continue
+			}
+			time.Sleep(1 * time.Second)
 			continue
 		}
-		logMessage := line.Text[secondSpaceIndex(line.Text):]
-		logChan <- logMessage
+		// exits
+		if err != nil {
+			break
```
Comment (mheon) [inline cmd/podman/logs.go]: Should we return this error? Maybe if it's not EOF return?

File: cmd/podman/logs.go (hunk 1 above)
Comment (baude) [reply cmd/podman/logs.go]: Yeah, thats the vndr stuff I keep forgetting to do

File: cmd/podman/logs.go (hunk 4 above)
Comment (baude) [reply cmd/podman/logs.go]: No, imho it is fine as is. The EOF is handled in follow and this handles ctrl-c etc.

File: cmd/podman/logs.go (hunk 2 above)
Comment (baude) [reply cmd/podman/logs.go]: same as above, imo it is fine as is

File: cmd/podman/logs.go (hunk 3 above)
Comment (baude) [reply cmd/podman/logs.go]: fixed

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
bot, retest this please

**Comment by mheon** [comment]
LGTM

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**